exception_safe_wrapper_test01